src/LiveComponent/src/ComponentWithFormTrait.php

```diff
@@ -161,14 +157,17 @@ private function submitForm(bool $validateAll = true): void
      * frontend, and it's meant to equal the raw POST data that would
      * be sent if the form were submitted without modification.
      */
-    private function extractFormValues(FormView $formView): array
+    private function extractFormValues(FormView $formView, array $values = []): array
     {
-        $values = [];
         foreach ($formView->children as $child) {
             $name = $child->vars['name'];
-            if (\count($child->children) > 0) {
-                $values[$name] = $this->extractFormValues($child);
+            if (!($child->vars['expanded'] ?? false) && \count($child->children) > 0) {
+                $values[$name] = $this->extractFormValues($child, $values[$name] ?? []);
+
+                continue;
+            }
 
+            if (null !== ($values[$name] ?? null)) {
                 continue;
             }
```

Comment:
What is this if statement for? If I comment it out, the tests still pass. So if it IS needed for something, we need to expand the test suite the cover it.

Reply:
It is here to make sure we don't overwrite changed fields with initial form data. I will surely write tests for that 👍

Reply:
On a second thought it might not work well with checkboxes that are initially checked. Noting for myself to take that situation into consideration within tests.

Reply:
In fact it wasn't needed there at all :)